src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -14,18 +15,57 @@ import scala.concurrent.{ExecutionContext, Future}
   * Classes used for deserializing query.
   */
 object DataTypes {
+
+  import cats.implicits._
   import io.scalaland.chimney.dsl._
 
   /** Type representing Map[String, Any] */
   type AnyMap = Map[String, Any]
 
   /** Type representing Map[String, Option[Any]] for query response */
   type QueryResponse = Map[String, Option[Any]]
+  /** Method checks if type can be aggregated */
+  lazy val canBeAggregated: DataType => Boolean = Set(DataType.Decimal, DataType.Int, DataType.LargeInt, DataType.DateTime)
   /** Default value of limit parameter */
   val defaultLimitValue: Int = 10000
   /** Max value of limit parameter */
   val maxLimitValue: Int = 100000
 
+  /** Helper method for finding fields used in query that don't exist in the database */
+  private def findNonExistingFields(query: Query, entity: String, tezosPlatformDiscovery: TezosPlatformDiscoveryOperations)
+    (implicit ec: ExecutionContext): Future[List[QueryValidationError]] = {
+    val fields = query.fields.map("query" -> _) ::: query.predicates.map("predicate" -> _.field) :::
```

Comment:
No need to change, just sharing ideas.
I find this is a perfect place to use scala Symbol, at least to me it reads better, thought technically there's no difference.

```scala
query.fields.map('query -> _)  :::
  query.predicates.map('predicate -> _.field) :::
  query.orderBy.map('orderBy -> _.field) :::
  [...]

And the same Symbol used to pattern match
```

Reply:
good point, changed